roxie/ccd/ccdprotocol.cpp

```diff
@@ -1523,12 +1533,13 @@ class RoxieSocketWorker : public ProtocolQueryWorker
     //  interface IPooledThread
     virtual void init(void *_r) override
     {
-        client.setown(new CSafeSocket((ISocket *) _r));
+        rawClient.setown((ISocket *) _r);
         ProtocolQueryWorker::init(_r);
     }
 
     virtual void threadmain() override
     {
+        client.setown(new CSafeSocket(rawClient.getClear()));
```

Comment:
Still do not think this runs the [potentially blocking] secure_accept() in the pooled thread ?
Perhaps:
client.setown(new CSafeSocket(createSecureSocket(rawClient.getClear());
or something like that ?  Handling exceptions etc.